### PR TITLE
store-histogram: skip not found file meta

### PR DESCRIPTION
**Pull request by HaoranYi**
#### Problem

store-histogram crash on not found file meta. 

```
thread 'main' panicked at accounts-db/store-histogram/src/main.rs:252:58:
called `Result::unwrap()` on an `Err` value: Os { code: 2, kind: NotFound, message: "No such file or directory" }
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

this can actually happen.
https://github.com/anza-xyz/agave/pull/2906#issuecomment-2347401279

#### Summary of Changes

instead of crash when file meta is not found, we skip them and continue.

Fixes #

### CODE REVIEW

File: accounts-db/store-histogram/src/main.rs [hunk 1]
```diff
@@ -249,7 +249,9 @@ fn main() {
             for entry in dir.flatten() {
                 if let Some(name) = entry.path().file_name() {
                     let name = name.to_str().unwrap().split_once(".").unwrap().0;
-                    let len = fs::metadata(entry.path()).unwrap().len();
+                    let len = fs::metadata(entry.path())
+                        .unwrap_or_else(|_| panic!("{:?} not found!", entry.path()))
```
Comment (brooksprumo) [inline accounts-db/store-histogram/src/main.rs]: How about something like this?
```suggestion
                        .unwrap_or_else(|err| panic!("failed to get metadata '{}': {err}", entry.path().display()))
```

Comment (brooksprumo) [reply accounts-db/store-histogram/src/main.rs]: You could also use [`.inspect_err()`](https://doc.rust-lang.org/std/result/enum.Result.html#method.inspect_err), and `.unwrap()`, but that may duplicate some of the displayed text.

**Review by brooksprumo — DISMISSED** [review]
:shipit:

**Comment by HaoranYi** [comment]
sorry to request review again. 

need to format the file.

**Comment by jeffwashington** [comment]
this is fine. I'd rather it just keep running and log failing files. If you are running this while a validator is running, it could clean something away and delete it after getting the dir but before opening the file.

**Comment by HaoranYi** [comment]
Ah. Make sense. That explains when I `watch` this command every 2 seconds. It failed from time to time. 
I push a commit to skip failed metadata.

**Review by brooksprumo — APPROVED** [review]
:shipit:

**Review by jeffwashington — APPROVED** [review]
lgtm

**Pull request merged**